File: src/bitstamp.cpp
#include "wsapi.hpp"
#include <iostream>

//
// subscribe to trades on bitstamp for btcusd
// create callback class derived from wsapi_cb
// print messages to std::cout

class callback : public wsapi_cb{
public:
  void recv(const std::string &msg) {
    std::cout << msg << std::endl;
  }
};

int main() {
  const char *host = "ws.bitstamp.net";
  const char *port = "443";
  const char *text = "{\"event\": \"bts:subscribe\", \"data\": {\"channel\": \"live_trades_btcusd\"}}";

  callback cb;
  wsapi ws(host, port, text, &cb);
  ws.start();
  return 0;
}

File: src/python/bitstamp.py
#
# listen to trades on bitstamp via websockets
import asyncio
import websockets
import json

address = "wss://ws.bitstamp.net"

async def listener(subscription):
    async with websockets.connect(address) as websocket:
        print("connected")
        await websocket.send(json.dumps(subscription))
        while True:
            try:
                response = await websocket.recv()
                print(response)
            except:
                print("Error Found")
                quit()
                

if __name__ == '__main__':
    subscription = {"event": "bts:subscribe", "data": {"channel": "live_trades_btcusd"}}
    print("running bitstamp listener on subscription {}".format(json.dumps(subscription)))
    asyncio.get_event_loop().run_until_complete(listener(subscription))

        
